main.cpp

```diff
@@ -168,138 +189,172 @@ struct CookieShop
-    //number of usb ports (int)
     int numOfUSBPorts = 6;
-    //number of keys (int)
     int numOfKeys = 120;
 
-    //display letters typed
     void typeLetter(bool isLetterKeyPressed);
-    //multiply numbers
-    int productOfNumbers(int num1, int num2);// returns product of numbers
-    //run software
+    int productOfNumbers(int num1, int num2);
     void runSoftware(bool hasDoubleClickedIcon);
 };
-/*
-Thing 4) Nuclear Power Plant
-5 properties:
-    1) amount of water used (gallons per mega-watt hr.) (float)
-    2) temp. of water (celcius) (float)
-    3) number of generators (int)
-    4) weight of generator (tons) (float)
-    5) number of engineers (int)
-3 things it can do:
-    1) produce heat
-    2) generate steam
-    3) produce electricity
- */
+
+void Laptop::typeLetter(bool isLetterKeyPressed)
+{
+    if (isLetterKeyPressed == true)
+    {
+        char letterPressed;
+        std::cin >> letterPressed;
+        std::cout << letterPressed << std::endl;
+    }
+}
+
+int Laptop::productOfNumbers(int num1, int num2)
+{
+    int product = num1 * num2;
+    return product;
+}
+
+void Laptop::runSoftware(bool hasDoubleClickedIcon)
+{
+    if (hasDoubleClickedIcon == true)
+    {
+        std::cout << "Software Running" << std::endl; 
+    }   
+}
+
 struct NuclearPowerPlant
 {
-    //amount of water used (gallons per mega-watt hr.) (float)
     float amountOfWaterUsed = 375.5f;
-    //temp. of water (celcius) (float)
     float tempOfWater = 284.8f;
-    //number of generators (int)
     int numOfGenerators = 3;
-    //weight of generator (tons) (float)
     float weightOfGenerators = 389.79f;
-    //number of engineers (int)
     int numOfEngineers = 100;
 
-    //produce heat
     void produceHeat(bool isFissionSuccessful);
-    //generate steam
     void generateSteam(bool tubesReachedNeededTemp);
-    //produce electricity
     void produceElectricity(bool isGeneratorReceivingSteam);
 };
-/*
-Thing 5) Airplane
-5 properties:
-    1) number of passenger seats (int)
-    2) total weight of plane (lbs.) (float)
-    3) length of plane wings (meters) (float)
-    4) number of luggage pieces (int)
-    5) amount of fuel (gallons) (int)
-3 things it can do:
-    1) take off and fly
-    2) land 
-    3) send data to control tower
- */
+
+void NuclearPowerPlant::produceHeat(bool isFissionSuccessful)
+{
+    if (isFissionSuccessful)
+    {
+        std::cout << "It's getting hot!" << std::endl;
+    }
+        std::cout << "Try again" << std::endl;
+}
+
+void NuclearPowerPlant::generateSteam(bool tubesReachedNeededTemp)
+{
+    if (tubesReachedNeededTemp)
+    {
+        std::cout << "Generating steam" << std::endl;
+    }
+        std::cout << "No steam generated" << std::endl;
+}
+
+void NuclearPowerPlant::produceElectricity(bool isGeneratorReceivingSteam)
+{
+    if (isGeneratorReceivingSteam)
+    {
+        std::cout << "Producing elctricity" << std::endl;
+    }
+        std::cout << " No electricity" << std::endl;
+}
```

Comment:
Fix this indentation
Code inside curly braces needs to be indented 4 spaces from those curly braces.
Also, your code will always output `No steam generated` and `No electricity` no matter what.
You must have forgotten to write `else { }` around those `std::cout` statements.
- [ ] revise